lib/ArrayUtils.js

```diff
@@ -218,6 +218,36 @@ class ArrayUtils {
     });
     return groups;
   }
+
+  /**
+   * Similar to the UNIX utility `uniq` and lodash's `_.sortedUniq`, returns `xs` with
+   * consecutive similar values removed.  Two values `x1`, `x2` are similar iff
+   * `key(x1) === key(x2)`.
+   *
+   * Note that this does not handle non-consecutive similar values.
+   *
+   * @param {Array} xs
+   * @param {ComparisonKey} key
+   * @see https://lodash.com/docs/4.17.15#sortedUniq
+   */
+  static consecutiveUniqBy(xs, key) {
```

Comment:
I _think_ it's still the right call to avoid a dependency on `lodash`, if only to help keep build times and bundle sizes manageable - but we should maybe revisit that decision later on if we keep adding more methods to `ArrayUtils`.